Bio/SeqIO/GckIO.py

```diff
@@ -20,11 +20,21 @@
 from .Interfaces import SequenceIterator
 
 
-def _read(handle, length):
-    """Read the specified number of bytes from the given handle."""
-    data = handle.read(length)
+class PrematureEndOfFileError(ValueError):
```

Comment:
Why do you feel this needs a new exception class, and should we not define and use this more centrally if it is worth adding? I suspect it could be used for several of the other binary file formats...

Reply:
Reading the next packet can fail because we have completed reading the file, or because of a premature end to the file. We need to distinguish these two, as the former should result in a `StopIteration` and the latter in a `ValueError`.